### ISSUE TITLE
User needs to login after page refresh in order to add event

### ISSUE THREAD

**Issue by aniketkambli** [Kubessandra/react-google-calendar-api#85]
The example is working fine but when i refresh the page, the google's session is not stored in browser and hence the user has to login again and provide consent before adding an event or else it throws error

**Comment by aniketkambli** [comment]
Also how to create events with custom start/end time and start/end date

**Comment by Kubessandra** [comment]
```javascript
  /**
    * Create calendar event
    * @param {string} CalendarId for the event by default use 'primary'.
    * @param {object} Event with start and end dateTime
    * @param {string} sendUpdates Acceptable values are: "all", "externalOnly", "none"
    * @returns {any} Promise on the event.
    */
   public createEvent(event: object, calendarId: string = this.calendar, sendUpdates: string = 'none',): any {
```

With this method you can use a start and end dateTime.

For the login issue, this was introduced with the new API of google where they do not manage the session state anymore.
I havn't found the time to find a solution yet.

**Comment by aniketkambli** [comment]
okay cool, let me know once you find a workaround for session persistence. Other than that great work on the package 👍

**Comment by MatiasZelayaDept** [comment]
Hi ! Any news about this issue? I have the same problem

**Comment by udaniketk** [comment]
Nah, still stuck on the same problem. Let me know if you find a workaround

**Comment by Kubessandra** [comment]
I havn't found the time to find a workaround with google yet..

**Comment by MatiasZelayaDept** [comment]
Maybe you can made a method to return the token and the refresh token and the user can handle that outside the library

**Comment by Kubessandra** [comment]
You already have access to it by using `apiCalendar.tokenClient`

https://developers.google.com/identity/oauth2/web/guides/use-token-model

**Comment by JanderSilv** [comment]
Any updates on this issue?

**Comment by vivek-opynion** [comment]
Any update on this issue.?

**Comment by UsamaEjaz0** [comment]
Any updates

**Comment by gabrielpagotto** [comment]
Any updates about this issue?

**Comment by Kubessandra** [comment]
Did not find the solution with the new Google API, if you have any idea, we can try to do a PR, but yes this is really annoying
